### ISSUE TITLE
Integration with Moodle

### ISSUE THREAD

**Issue by oyara** [microsoft/Learn-LTI#71]
I have created a new environment in Azure and follow Deployment Guide <-- All Ok
When I try to configure Moodle (LTI 1.1) I can't see the following option in Step 7
![image](https://user-images.githubusercontent.com/1382584/94944207-36477880-04d9-11eb-9462-25793f639aae.png)
![image](https://user-images.githubusercontent.com/1382584/94944770-0056c400-04da-11eb-9feb-6dc29397e7d8.png)

I have completed all the configuration steps (without step 7) and then I create a new activity following the instructions. When Azure window opens I get this error screen
![image](https://user-images.githubusercontent.com/1382584/94944348-6c84f800-04d9-11eb-8ca1-84423f5876b8.png)
and I can see this using Developer Tools
![image](https://user-images.githubusercontent.com/1382584/94944496-a0f8b400-04d9-11eb-9a72-07adea60bcb1.png)

What can I do to solve this issue?

**Comment by adbindal** [comment]
Hello @oyara,

Thanks a lot for trying **Learn-LTI Tool** in your environment. Based on the above inputs, there are a few things, listed below, that I believe might be of some help in your case.

### 1. Missing Name Role Provisioning Service in LTI1.1
_NRPS_ is something that has been a part of _LTIAdvantage_ Specification which, per my knowledge, is not available in LTI1.1 based Moodle. So, you've done the right thing to move ahead without doing anything on this front.

### 2. Unable to open the Assignments page in Learn-LTI
From the screenshot, it feels like the `users-xxxxx/{id}/me` called failed with `http500` which essentially has been the source of the **_Oops!_** error message. I've shared some insights on debugging such User API failures in #55. I'd request you to please take a look at the same and see if the inputs shared there could be of some help in your case as well.

**Comment by oyara** [comment]
Thanks a lot @adbindal you are correct. I have errors in that function getting user information of my user
![image](https://user-images.githubusercontent.com/1382584/94947656-4c0b6c80-04de-11eb-995f-55c1a7c6645d.png)
I'll send my user (e-mail) by e-mail to [learnlti@microsoft.com](learnlti@microsoft.com)

**Comment by oyara** [comment]
I can't send messages to learnlti@microsoft.com. Sorry the message is in Spanish.

![image](https://user-images.githubusercontent.com/1382584/94948095-08fdc900-04df-11eb-86d9-10f5ca1cc936.png)

How can I share my e-mail address involved in Azure Function?
Thanks a lot.

**Comment by adbindal** [comment]
Hi @oyara 

Thanks for the information. Can you please confirm that the user with which you are trying to sign into AAD is actually enrolled in the course? (See https://github.com/microsoft/Learn-LTI/issues/62#issuecomment-699803160 for more information on this).

Regarding the inability to share the logs via the shared email, let me circle back with the team on the updated email and get back to you in this thread itself.

**Comment by oyara** [comment]
Thanks a lot @adbindal. I'll check  exceptions

**Comment by oyara** [comment]
Hello again @adbindal 
Related with Exceptions I can see this exception several times

Browser's console
![image](https://user-images.githubusercontent.com/1382584/94990330-8840da80-057b-11eb-8895-6149972d370e.png)

Application Insights
![image](https://user-images.githubusercontent.com/1382584/94990298-4adc4d00-057b-11eb-9d71-5266d467c690.png)

I have confirmed that my account is enrolled in the course as Manager and Teacher

How can I continue? 

Thanks a lot.

**Comment by adbindal** [comment]
> An Update on the shared email, we've updated the email id and you should be able to reach us out on learnlti-support@microsoft.com. We'll connect with you to get you unblocked with the tool usage.

Based on your newer inputs, the failure seems to be happening at the time of getting the assignment information when trying to fetch user details. This is required since it is the `Assignment` that holds the context membership url. While I don't have much leads to pin-point the failure, it seems like there's some issue when trying to fetch the assignment from its storage. There are 2 points to be noted here: -
1. We'd need to understand whether the assignment is present in the storage or not. Since, in the absence of the assignment, `GetUserDetails` API is expected to fail for sure. 
2. It seems like there's a missing null check in the code as well, which is causing the crash to occur. We'll scrub this on our end, as we have necessary code in place to return `http204` in case of a missing assignment from storage.

To dig down further on the exceptions, you can click them on the left hand and they would unravel their details on the right. This way you should have access to the complete crash call stack. Combining that information with code should help us arrive at a possible

**Comment by oyara** [comment]
Hello @adbindal 
I have checked Function CreateOrUpdateAssignment and it works ok

![image](https://user-images.githubusercontent.com/1382584/95000735-97e41180-05c3-11eb-8f39-0b442ca8790d.png)

Last execution details:

![image](https://user-images.githubusercontent.com/1382584/95000744-afbb9580-05c3-11eb-92a1-be582af70158.png)

First part of the error in GetUserDetails:

![image](https://user-images.githubusercontent.com/1382584/95000787-340e1880-05c4-11eb-9d58-abf857126fed.png)

![image](https://user-images.githubusercontent.com/1382584/95000791-438d6180-05c4-11eb-83c4-b8c27f17ee7d.png)

Microsoft.Azure.WebJobs.Host.FunctionInvocationException:
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor+<ExecuteWithLoggingAsync>d__18.MoveNext (Microsoft.Azure.WebJobs.Host, Version=3.0.22.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35Microsoft.Azure.WebJobs.Host, Version=3.0.22.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35: C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.csMicrosoft.Azure.WebJobs.Host, Version=3.0.22.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35: 287)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor+<TryExecuteAsync>d__15.MoveNext (Microsoft.Azure.WebJobs.Host, Version=3.0.22.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35Microsoft.Azure.WebJobs.Host, Version=3.0.22.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35: C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.csMicrosoft.Azure.WebJobs.Host, Version=3.0.22.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35: 92)
Inner exception System.NullReferenceException handled at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw:
   at LtiLibrary.NetCore.Clients.MembershipClient.GetFilteredServiceUrl (LtiLibrary.NetCore, Version=3.1.1.0, Culture=neutral, PublicKeyToken=null)
   at LtiLibrary.NetCore.Clients.MembershipClient+<GetMembershipAsync>d__0.MoveNext (LtiLibrary.NetCore, Version=3.1.1.0, Culture=neutral, PublicKeyToken=null)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at Edna.Bindings.Lti1.Lti1MembershipClient+<GetAllMembers>d__4.MoveNext (Edna.Bindings.Lti1, Version=1.0.0.0, Culture=neutral, PublicKeyToken=nullEdna.Bindings.Lti1, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null: C:\Users\angel\Learn-LTI\backend\Bindings\Edna.Bindings.Lti1\Edna.Bindings.Lti1\Lti1MembershipClient.csEdna.Bindings.Lti1, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null: 34)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at Edna.Bindings.Lti1.Lti1MembershipClient+<GetMemberByEmail>d__5.MoveNext (Edna.Bindings.Lti1, Version=1.0.0.0, Culture=neutral, PublicKeyToken=nullEdna.Bindings.Lti1, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null: C:\Users\angel\Learn-LTI\backend\Bindings\Edna.Bindings.Lti1\Edna.Bindings.Lti1\Lti1MembershipClient.csEdna.Bindings.Lti1, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null: 48)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at Edna.Users.UsersApi+<GetUserDetails>d__5.MoveNext (Edna.Users, Version=1.0.0.0, Culture=neutral, PublicKeyToken=nullEdna.Users, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null: C:\Users\angel\Learn-LTI\backend\Functions\Edna.Users\Edna.Users\UsersApi.csEdna.Users, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null: 68)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionInvoker`2+<InvokeAsync>d__10.MoveNext (Microsoft.Azure.WebJobs.Host, Version=3.0.22.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35Microsoft.Azure.WebJobs.Host, Version=3.0.22.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35: C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionInvoker.csMicrosoft.Azure.WebJobs.Host, Version=3.0.22.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35: 52)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor+<InvokeAsync>d__26.MoveNext (Microsoft.Azure.WebJobs.Host, Version=3.0.22.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35Microsoft.Azure.WebJobs.Host, Version=3.0.22.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35: C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.csMicrosoft.Azure.WebJobs.Host, Version=3.0.22.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35: 539)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor+<ExecuteWithWatchersAsync>d__25.MoveNext (Microsoft.Azure.WebJobs.Host, Version=3.0.22.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35Microsoft.Azure.WebJobs.Host, Version=3.0.22.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35: C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.csMicrosoft.Azure.WebJobs.Host, Version=3.0.22.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35: 486)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor+<ExecuteWithLoggingAsync>d__24.MoveNext (Microsoft.Azure.WebJobs.Host, Version=3.0.22.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35Microsoft.Azure.WebJobs.Host, Version=3.0.22.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35: C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.csMicrosoft.Azure.WebJobs.Host, Version=3.0.22.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35: 430)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor+<ExecuteWithLoggingAsync>d__18.MoveNext (Microsoft.Azure.WebJobs.Host, Version=3.0.22.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35Microsoft.Azure.WebJobs.Host, Version=3.0.22.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35: C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.csMicrosoft.Azure.WebJobs.Host, Version=3.0.22.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35: 239)

It seems that assignment is created :-) 

Let me know if I can share additional information or I need to do some additional steps.

Thanks for your help

**Comment by adbindal** [comment]
Thank You for sharing more insights on the `NullException` and your patience on the issue. 

With the new information, I believe I have a fairly elaborate idea on what might be happening in your environment. While I place the details below, I'd like to work closely with you and dig further in your environment to understand the reasons of the below behavior. Request you to please reach us out on learnlti-support@microsoft.com so that we can help you out in a better manner.

### Workflow Diagnostic
1. Upon the call to `GetUsersDetails`, a server-to-server call is made to `GetAssignment` to get the `ContextMembershipUrl` and other details related to the current `assignmentId`. Unfortunately, inside that call, there's a failure in fetching the `AssignmentEntity` from the corresponding `CloudTable`. This is what is indicated as `http409` in the above screenshot.
2. The failure ends up filling the `AssignmentDto` with incorrect values. This skips our `null` checks and returns `http200` to `GetUsersDetails`.
3. `GetUsersDetails` tries to use members, like `ContextMembershipUrl` , of this `AssignmentDto` and pass them over to `LtiLibrary`. Due to lack of some error handling in this workflow, we end up using a `null` member of `AssignmentDto`, causing the 2nd set of exceptions we see in above screenshot, being able to correlate using the crash stack.

### Current State

* While points 2 and 3 are something we can fix in the code, but unfortunately that would not get your environment functional until `CloudTable.ExecuteAsync` keeps raising `http409` (conflict). 
* The confusing part is that the query made is for "getting" the information from table and it is not quite evident why **Azure Storage Table** would raise a Conflict on the such query based on the existing documentations. 
* Maybe one of the things to check is to make a call to `GetAssignment` via postman to ensure that this API is working via some invocation. Another source of some information might be in `GetAssignment` azure function inside `assignments-xxxx` function-app.

**Comment by oyara** [comment]
Hello @adbindal, the function GetAssignment is working perfect:

![image](https://user-images.githubusercontent.com/1382584/95110695-b7ec1000-073e-11eb-8fad-0ea58aff8da4.png)

I'll send you an e-mail and we can continue solving the issue.

Thanks for your help again.

**Comment by adbindal** [comment]
Thanks @oyara for the details and reaching out to us for a quick deep-dive on the issue. Here are a few takeaways from the discussion, please feel free to add / update anything from your end.

1. We were able to isolate the issue being that Moodle is not passing us the **custom_context_membership_url** key as a part of the _LTI1-Launch_ api payload. Without this url, the Learn-LTI tool is unable to query Moodle for course participants and thus unable to verify that the assignment user is actually enrolled in the course. This leads to **_http500_** returned from _GetUserDetails_ call.
2. **Action Items:** We checked that the deployed moodle is of version 3.3.1+ and this is something on which we’ve not yet tested our Learn-LTI implementation. As next steps, we planned on the following items w.r.t moodle front: -
   1. Create a support ticket with Moodle to understand their stand w.r.t supporting _ContextMembership_ API in Moodle **v3.3.1**.
   2. Consider upgrading Moodle version. The one we are using is **v3.8.5**, and that supports both _LTI1.1_ and _LTI1.3_.
   3. Probably look out for any Moodle plugin or a hidden setting that allows the _**LTI Names and Role Provisioning setting**_ to show up in Tool Configuration menu (see image below).
 3. We agreed to keep the issue open until there's a resolution of this from your end. Your inputs on the issue should deeply help future adopters of the our Learn-Lti tool who find themselves in similar problem.

Thanks again for all your patience with issue and trying to use onboard our Tool in your environment.

---
![image](https://user-images.githubusercontent.com/5679830/95218474-edadf900-0811-11eb-97d1-aff43bf8b6bc.png)

> Apologies for my first reply regarding this setting, while I was of the opinion that absence of this setting should not be much problem, it seems like its absence is the actual problem.

**Comment by dcgchico** [comment]
Hello @oyara
I have the same problem as you (exactly the same). Finally, did you solve it?

**Comment by oyara** [comment]
> Hello @oyara
> I have the same problem as you (exactly the same). Finally, did you solve it?

Hello @dcgchico . We have updated our Moodle platform and next week I will check that the plugin is working. I'll update here with the new information.

**Comment by leestott** [comment]
Hi @oyara @dcgchico did you resolve the issues with a upgrade of your Moodle environments?

Can this issue be closed or updated to reflect any changes you have made?

**Comment by leestott** [comment]
@oyara @dcgchico We think we identified the issue as its simply related to students/teachers not being enrolled and active on a course 
All users need to be active on a course if you see Not Current next to the status.. 

![image](https://user-images.githubusercontent.com/2511341/108256622-a3406f80-7155-11eb-9ef2-b9b42faf324f.png)

We have noticed that users having this issue 
https://user-images.githubusercontent.com/1382584/94944496-a0f8b400-04d9-11eb-9a72-07adea60bcb1.png

have students/teacher who may be manually assigned or automatically assigned as showing  "Not Current". 

The solution is simply make the students and teachers active and they can then log in. 

To activate users 
Click on the settings wheel in the top right hand corner and select "Enrollment methods" then click the crossed out eye to allow manual enrollment. 

When you go back to the course participants they will show as active. 

Hope this helps.

**Comment by dcgchico** [comment]
Hi @leestott,
We continue with the problem. We have Moodle 3.8.5+ (Build: 20201028)
We receive the error "You are not enrolled in this course" when we use LTI 1.1. If I configure the course using LTI 1.3, I receive the error "Oops! Something went wrong." (and the same error as @oyara in function GetUserDetails: "Could not get members")
My users are active, so I think that's not the problem.
![image](https://user-images.githubusercontent.com/72253266/108340111-8305c480-71d8-11eb-8db6-f3c7adb9ee69.png)

**Comment by leestott** [comment]
@dcgchico can you please send a email to learnlti@microsoft.com including you Azure functions debug and screenshot of the error. Can you confirm the users email address in Moodle also matches the email in AAD and you have logged in with a teacher account first to setup the course.

**Comment by leestott** [comment]
@dcgchico @oyara can you confirm if your still experiencing issues the troubleshooting guide has been updated to now include documented fixes for your issues. Can we now close these issues.

**Comment by dcgchico** [comment]
The problem still remains. We are in the case [Error Unauthorized 401 when using Moodle](https://github.com/microsoft/Learn-LTI/blob/main/docs/TROUBLESHOOTING.md#error-unauthorized-401-when-using-moodle), but we are using nginx server, not Apache, so we can't apply de apache config solution provided

**Comment by leestott** [comment]
@dcgchico please email learnlti@microsoft.com with screenshots and we can schedule a meeting to look at this issue.

**Comment by leestott** [comment]
@dcgchico Hi David did the above help resolve the blocker using nginx?

**Comment by dcgchico** [comment]
We made the changes you referred, but the error is the same. Is there any log in our moodle server where we can see anything?

**Comment by leestott** [comment]
Closing this issue as no updates received from April please open a new issue if issues are still present.
